lib/mistral_ai_summary_example/settings_model.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:flutter/material.dart';
+
+class SummarySettings {
+  const SummarySettings({
+    this.model = MistralAIModel.mistralSmall,
+    this.temperature = 0.7,
+    this.topP = 1,
+    this.maxTokens,
+    this.safePrompt = false,
+    this.randomSeed,
+  });
+  final MistralAIModel model;
+  final double temperature;
+  final double topP;
+  final int? maxTokens;
+  final bool safePrompt;
+  final int? randomSeed;
+}
+
+enum MistralAIModel {
+  mistralMedium('mistral-medium'),
+  mistralSmall('mistral-small'),
+  mistralTiny('mistral-tiny'),
+  mistralEmbed('mistral-embed');
+
+  const MistralAIModel(this.name);
+  final String name;
+}
+
+class SummarySettingsModel extends ChangeNotifier {
```

Comment:
I think you could use `ValueNotifier` which will automatically update whenever you change the `value` of it. This will save you from calling `notifyListeners()` every time you want to update a setting but require you to have a `copyWith` function in `SummarySettings`.

Reply:
Because we don't use any package that can generate `copyWith` I prefer to use `notifyListeners` to update the settings.

Reply:
Let's leave it for now. It's not that important.